tests/test_flow/test_copy.py

```diff
@@ -50,6 +51,40 @@ def dask_df():
     return builder.build()
 
 
+@pytest.fixture
```

Comment:
This is an interesting special case! I agree that we might as well handle this with patching. However, there a couple things about your implementation I don't understand:

1. Why do we need to return a context manager instead of doing the patching inside the fixture like we (used to) do elsewhere? I.e.,

    do_patching()
    yield None
    undo_patching()

2. Do we still need separate methods for `_upload_to_gcs` and `upload_to_gcs`? Could we accomplish the same thing with just a global `gcs_fs` object (like `_cached_gcs_fs`) that we patch out?